modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc

```diff
@@ -39,35 +41,60 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
 
 Status PathReuseDecider::Process(Frame* const frame,
                                  ReferenceLineInfo* const reference_line_info) {
+  if (!Decider::config_.path_reuse_decider_config().reuse_path())
+    return Status::OK();
   // Sanity checks.
   CHECK_NOTNULL(frame);
   CHECK_NOTNULL(reference_line_info);
 
-  // do not reuse path during sidepass
-  if (Decider::config_.path_reuse_decider_config().reuse_path() &&
-      !reference_line_info->is_path_lane_borrow()) {
-    ++total_path_counter_;  // count total path
+  // check front static blocking obstacle
+  auto* mutable_path_reuse_decider_status = PlanningContext::Instance()
+                                                ->mutable_planning_status()
+                                                ->mutable_path_reuse_decider();
+  auto* mutable_path_decider_status = PlanningContext::Instance()
+                                          ->mutable_planning_status()
+                                          ->mutable_path_decider();
+  const int kWaitCycle = -2;  // wait 2 cycle
+  ADEBUG << "reuse or not: "
+         << mutable_path_reuse_decider_status->reused_path();
+  if (mutable_path_reuse_decider_status->reused_path()) {
+    ADEBUG << "reused path";
     if (CheckPathReusable(frame, reference_line_info)) {
+      ADEBUG << "no collision";
       ++reusable_path_counter_;  // count reusable path
       if (!TrimHistoryPath(frame, reference_line_info))
-        AERROR << "Failed to trim reference line ";
+        AERROR << "Failed to trim reused path";
+    } else {
+      // disable reuse path
+      ADEBUG << "collision!!!";
+      mutable_path_reuse_decider_status->set_reused_path(false);
+    }
+  } else {
+    ADEBUG
+        << "counter: "
+        << mutable_path_decider_status->front_static_obstacle_cycle_counter();
+    if (mutable_path_decider_status->front_static_obstacle_cycle_counter() <
+        kWaitCycle) {
+      // enable reuse path
+      mutable_path_reuse_decider_status->set_reused_path(true);
     }
   }
+  ++total_path_counter_;
   ADEBUG << "reusable_path_counter_" << reusable_path_counter_;
   ADEBUG << "total_path_counter_" << total_path_counter_;
   return Status::OK();
 }
 
 bool PathReuseDecider::CheckPathReusable(
     Frame* const frame, ReferenceLineInfo* const reference_line_info) {
-  if (!IsSameStopObstacles(frame, reference_line_info))
-    ADEBUG << "not same stop obstacle";
-  return (IsCollisionFree(reference_line_info) &&
-          IsSameStopObstacles(frame, reference_line_info));
+  //   if (!IsSameStopObstacles(frame, reference_line_info))
+  //     ADEBUG << "not same stop obstacle";
+  //   return (IsCollisionFree(reference_line_info) &&
+  //           IsSameStopObstacles(frame, reference_line_info));
   //   return IsSameStopObstacles(frame, reference_line_info);
   //   return IsSameObstacles(reference_line_info);
```

Comment:
Either remove unused code, or add `TODO` to state why it's not used, which `return` will be used in what kind of situation. This piece of code is likely to be unmaintainable very soon.